repo/content/index/index_v2.go

```diff
@@ -136,96 +135,63 @@ type indexV2FormatInfo struct {
 	encryptionKeyID     byte
 }
 
-type indexV2EntryInfo struct {
-	data      []byte
-	contentID ID
-	b         *indexV2
-}
-
-func (e indexV2EntryInfo) GetContentID() ID {
-	return e.contentID
-}
-
-func (e indexV2EntryInfo) GetTimestampSeconds() int64 {
-	return int64(decodeBigEndianUint32(e.data[v2EntryOffsetTimestampSeconds:])) + int64(e.b.hdr.baseTimestamp)
-}
-
-func (e indexV2EntryInfo) GetDeleted() bool {
-	return e.data[v2EntryOffsetPackOffsetAndFlags]&v2EntryDeletedFlag != 0
-}
-
-func (e indexV2EntryInfo) GetPackOffset() uint32 {
-	return decodeBigEndianUint32(e.data[v2EntryOffsetPackOffsetAndFlags:]) & v2EntryPackOffsetMask
+type indexV2 struct {
+	hdr         v2HeaderInfo
+	data        []byte
+	closer      func() error
+	formats     []indexV2FormatInfo
+	packBlobIDs []blob.ID
```

Comment:
👍🏼 Memoizing the pack ids makes sense to avoid unnecessary frequent string allocations 🎉